c2rust-analyze/src/main.rs

```diff
@@ -150,6 +150,18 @@ fn run(tcx: TyCtxt) {
         gacx.fn_sigs.insert(ldid.to_def_id(), lsig);
     }
 
+    // Label the field types of each struct.
+    for ldid in tcx.hir_crate_items(()).definitions() {
+        let did = ldid.to_def_id();
+        if tcx.def_kind(did) != DefKind::Struct {
+            continue;
+        }
```

Comment:
Maybe just remove this check and always use `.all_fields()` (superseding my previous comments if we do it this way).  That way anything should hopefully be handled, like variant fields.

Reply:
If we don't do it this way, then to keep parity with `adt_func`, I'd use `.variant(0usize.into()).fields` instead of `.all_fields()` again here.

Reply:
I think we do need this check so we don't call `adt_def` on non-ADTs (I don't know of a way to enumerate only ADTs in the crate, only a way to enumerate all items).

Reply:
Is there a way to expand this check to all ADTs, not just structs?

Reply:
does cc7d9c5c cover all possibilities?

Reply:
I don't think `Variant`s have their own `AdtDef`s - they only exist as part of an enum.  If you run this on code that defines an enum, then I'd expect it to panic on the `adt_def` call below.  Only `Struct`, `Enum`, and `Union` should be accepted here.

Reply:
Ok, fixed in https://github.com/immunant/c2rust/commit/3e040e817d91b8ab14f4247f5f54e431e9ce8411